changelogs/fragments/2024-module-helper-fixes.yml

```diff
@@ -0,0 +1,5 @@
+bugfixes:
+  - module_helper module utils - actually ignoring formatting of parameters with value ``None`` (https://github.com/ansible-collections/community.general/pull/2024).
+  - module_helper module utils - handling ``ModuleHelperException`` now properly calls ``fail_json()`` (https://github.com/ansible-collections/community.general/pull/2024).
+minor_changes:
+  - module_helper module utils - use the command name as-is in ``CmdMixin`` if it fails ``get_bin_path()`` - allowing full path names to be passed (https://github.com/ansible-collections/community.general/pull/2024).
```

Comment:
Isn't this also a bugfix? If it failed before, it would have crashed, wouldn't it?

Reply:
(If this is a bugfix: this can be backported to stable-1 as well, right? Assuming the module helper is already present there, I don't remember anymore :) )

Reply:
Yes, it is present there.

Technically this is a new feature, put in place to support `cpanm` module parameter `executable` in which the user may specify the full path of the `cpanm` script. But if you don't mind bending the perspective a little bit, we could see that problem as a bug, of course. Please let me know if you want me to put another PR for this one.

Reply:
Right now, `get_bin_path` will return `None` if it cannot find the command, and this function will thus replace `cmd_args[0]` by `None`, which should cause `module.run_command` to crash. From that POV, this is clearly a bugfix and not a feature :)

Reply:
OK, fair enough :-)